sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryStorageQuerySource.java

```diff
@@ -57,6 +59,32 @@
         priority,
         location,
         kmsKey,
+        format,
+        parseFn,
+        outputCoder,
+        bqServices);
+  }
+
+  public static <T> BigQueryStorageQuerySource<T> create(
```

Comment:
Do users depend on these constructors in any way? We don't want to break backwards compatibility.

Reply:
Good catch. The whole class is still experimental, so changes should be OK to make. On the other hand, I wonder if the sources should be marked `@Internal` or even made non-public with only the transform being public.

Reply:
+1 for making this entire source non-public unless we actually intend for users to create one themselves. 

It seems like there is some precedent for making BoundedSource implementations public (AvroSource, ElasticSearchIO.BoundedElasticsearchSource), but maybe that's something that should be cleaned up.

In the elasticsearch case at least the source is annotated VisibleForTesting, but all the tests are in the same package, so I don't think this is necessary?

Reply:
I have also noticed that `BigQueryStorageStreamSource` is public as well. Will change it to package-private as well as `BigQueryStorageQuerySource`.